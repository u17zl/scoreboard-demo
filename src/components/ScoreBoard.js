import React, { Fragment, useState, useEffect } from "react";
import { useSpring, animated } from "react-spring";
import { MainSection, TeamStat } from "./components";

const ScoreBoard = props => {
  const { home, away, score, stat, animation } = props;
  const [mainShow, setMainShow] = useState(false);
  const [statShow, setStatShow] = useState(false);

  //   input error handler
  useEffect(() => {
    switch (animation.length) {
      case 0:
        setMainShow(false);
        setStatShow(false);
        break;

      case 1:
        if (animation[0]["animation"] === "teamStat") {
          break;
        }
        console.log("hello");
        setMainShow(true);
        setStatShow(false);
        break;

      case 2:
        if (
          animation[0]["animation"] === "teamStat" &&
          animation[1]["animation"] === "main"
        ) {
          setMainShow(true);
          setStatShow(true);
          break;
        } else if (
          animation[1]["animation"] === "teamStat" &&
          animation[0]["animation"] === "main"
        ) {
          setMainShow(true);
          setStatShow(true);
          break;
        }

      default:
        break;
    }
  }, [props]);

  return (
    <Fragment>
      {console.log(mainShow)}
      {mainShow ? (
        <MainSection home={home} away={away} score={score} />
      ) : (
        <div style={{ height: 21 }} />
      )}
      {statShow ? <TeamStat stat={stat} /> : null}
    </Fragment>
  );
};

export default ScoreBoard;
