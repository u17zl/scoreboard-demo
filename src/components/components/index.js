export { default as MainSection } from "./MainSection";
export { default as TeamStat } from "./TeamStat";
