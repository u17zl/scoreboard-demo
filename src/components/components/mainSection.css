.main-home {
  line-height: 2rem;
  font-size: 1.5rem;
  width: 10%;
  text-align: center;
  color: #fff;
  font-weight: 700;
}
.main-score {
  line-height: 2rem;
  font-size: 2rem;
  text-align: center;
  color: #fff;
  font-weight: 700;
}
.main-away {
  line-height: 2rem;
  font-size: 1.5rem;
  text-align: center;
  color: #000;
  width: 10%;
  font-weight: 700;
}
.main-time {
  line-height: 2rem;
  font-size: 1.5rem;
  width: 10%;
  text-align: center;
  color: #fff;
  font-weight: 700;
}
