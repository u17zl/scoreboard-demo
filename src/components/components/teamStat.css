.team-stat {
  display: inline-block;
  width: 50%;
  text-overflow: ellipsis;
  white-space: nowrap;
  line-height: 2rem;
  font-size: 1.5rem;
  text-align: center;
  color: #fff;
  font-weight: 700;
}
.info-stat {
  display: inline-block;
  width: 50%;
  text-overflow: ellipsis;
  white-space: nowrap;
  line-height: 2rem;
  font-size: 1.5rem;
  text-align: center;
  color: #fff;
  font-weight: 500;
}
