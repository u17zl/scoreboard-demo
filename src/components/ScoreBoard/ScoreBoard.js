import React, { Fragment, useState, useEffect } from "react";
import { MainSection, TeamStat } from "./components";

const ScoreBoard = props => {
  const { home, away, score, stat, animation } = props;
  const [mainShow, setMainShow] = useState(false);
  const [statShow, setStatShow] = useState(false);
  const [isBoth, setIsBoth] = useState(false);
  const [mainDelay, setMainDelay] = useState(0);
  const [statDelay, setStatDelay] = useState(0);

  //   input error handler
  useEffect(() => {
    switch (animation.length) {
      case 0:
        setIsBoth(true);
        setStatShow(false);
        setMainShow(false);
        break;

      case 1:
        if (animation[0]["animation"] === "teamStat") {
          break;
        }
        setIsBoth(false);
        setMainShow(true);
        setStatShow(false);

        setMainDelay(animation[0]["delay"]);
        break;

      case 2:
        if (
          animation[0]["animation"] === "teamStat" &&
          animation[1]["animation"] === "main"
        ) {
          setIsBoth(false);
          setMainShow(true);
          setStatShow(true);
          setMainDelay(animation[1]["delay"]);
          setStatDelay(animation[0]["delay"]);
          break;
        } else if (
          animation[1]["animation"] === "teamStat" &&
          animation[0]["animation"] === "main"
        ) {
          setIsBoth(false);
          setMainShow(true);
          setStatShow(true);
          setMainDelay(animation[0]["delay"]);
          setStatDelay(animation[1]["delay"]);
          break;
        }

      default:
        break;
    }
  }, [props, isBoth]);

  return (
    <Fragment>
      <div className="board-grid">
        <MainSection
          show={mainShow}
          statShow={statShow}
          home={home}
          away={away}
          score={score}
          statShow={statShow}
          delay={mainDelay * 1000}
        />
        <TeamStat
          show={statShow}
          stat={stat}
          home={home}
          away={away}
          delay={statDelay * 1000 + mainDelay * 1000}
          isBoth={isBoth}
        />
      </div>
    </Fragment>
  );
};

export default ScoreBoard;
