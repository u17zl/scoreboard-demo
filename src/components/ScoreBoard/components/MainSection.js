import React, { Fragment } from "react";
import { PoseGroup } from "react-pose";
import "../styles/mainSection.scss";
import * as mainAnim from "../animation/mainSectionAnim";

const {
  Home,
  HomeText,
  HomeBorder,
  Score,
  ScoreText,
  Away,
  AwayBorder,
  AwayText,
  Time,
  TimeText
} = mainAnim;

const MainSection = props => {
  const { show, statShow, home, away, score, delay } = props;

  const {
    color: homeColor,
    name: homeName,
    abbr: homeAbbr,
    borderColor: homeBorderColor
  } = home;
  const {
    color: awayColor,
    name: awayName,
    abbr: awayAbbr,
    borderColor: awayBorderColor
  } = away;
  const { home: homeScore, away: awayScore } = score;

  return (
    <Fragment>
      <PoseGroup>
        {show && [
          <Home
            key="home"
            className="main-home"
            style={{ backgroundColor: homeColor }}
            delay={delay}
          >
            <HomeText key="texthome" delay={delay}>
              {homeAbbr}
            </HomeText>
            <HomeBorder
              key="borderhome"
              className="border-home"
              color={homeBorderColor}
              delay={delay}
            />
          </Home>,
          <Score
            key="score"
            className="main-score"
            style={{ backgroundColor: "#001339" }}
            delay={delay}
          >
            <ScoreText
              key="textscore"
              delay={delay}
            >{`${homeScore}-${awayScore}`}</ScoreText>
          </Score>,
          <Away
            key="away"
            className="main-away"
            style={{ backgroundColor: awayColor }}
            delay={delay}
          >
            <AwayBorder
              key="borderaway"
              className="border-away"
              color={awayBorderColor}
              delay={delay}
            />
            <AwayText key="textaway" delay={delay}>
              {awayAbbr}
            </AwayText>
          </Away>
        ]}

        {show && !statShow && (
          <Time
            key="time"
            className="main-time"
            style={{ backgroundColor: "#001339" }}
            delay={delay}
          >
            <TimeText key="time-text">90:00</TimeText>
          </Time>
        )}
      </PoseGroup>
    </Fragment>
  );
};

export default MainSection;
