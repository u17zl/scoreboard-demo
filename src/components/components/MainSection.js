import React, { Fragment } from "react";
import "./mainSection.css";
import { useSpring, animated } from "react-spring";

const MainSection = props => {
  const { show } = props;
  const homeColor = "red";
  const awayColor = "white";
  const homeAnim = useSpring({});
  const awayAnim = useSpring({});
  const fade = useSpring({
    from: {
      opacity: 0
    },
    opacity: 1
  });

  return (
    <Fragment>
      <animated.div
        className="main-home"
        style={{ backgroundColor: `${homeColor}` }}
      >
        KC
      </animated.div>
      <animated.div className="main-score" style={{ backgroundColor: "black" }}>
        0-4
      </animated.div>
      <animated.div
        className="main-away"
        style={{ backgroundColor: `${awayColor}` }}
      >
        GG
      </animated.div>
    </Fragment>
  );
};

export default MainSection;
