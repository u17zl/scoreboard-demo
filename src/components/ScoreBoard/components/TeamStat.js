import React, { Fragment, useState, useEffect } from "react";
import "../styles/teamStat.scss";
import posed, { PoseGroup } from "react-pose";
import * as statAnim from "../animation/statSectionAnim";

const { Team, TeamText, Stat, StatText } = statAnim;
const TeamStat = props => {
  const { show, home, away, stat, delay, isBoth } = props;
  const [statData, setSataData] = useState({ name: "", stat: "", color: "" });
  const { color: homeColor, name: homeName, abbr: homeAbbr } = home;
  const { color: awayColor, name: awayName, abbr: awayAbbr } = away;

  useEffect(() => {
    if (Object.keys(stat)[0] === "home") {
      setSataData({ name: homeName, stat: stat["home"], color: homeColor });
    } else {
      setSataData({ name: awayName, stat: stat["away"], color: awayColor });
    }
  }, [props]);

  return (
    <Fragment>
      <PoseGroup>
        {show && [
          <Team
            key="stat"
            className="team-stat"
            style={{ backgroundColor: statData.color }}
            delay={delay}
            isBoth={isBoth}
          >
            <TeamText delay={delay} isBoth={isBoth}>
              {statData.name}
            </TeamText>
          </Team>,
          <Stat
            key="stat-info"
            className="info-stat"
            style={{ backgroundColor: "#001339" }}
            delay={delay}
            isBoth={isBoth}
          >
            <StatText key="stat-text" delay={delay} isBoth={isBoth}>
              {statData.stat}
            </StatText>
          </Stat>
        ]}
      </PoseGroup>
    </Fragment>
  );
};

export default TeamStat;
