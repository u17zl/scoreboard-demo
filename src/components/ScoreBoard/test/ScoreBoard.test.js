import React from "react";
import Enzyme, { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";
import ScoreBoard from "../ScoreBoard";
import { MainSection, TeamStat } from "../components";
import Adapter from "enzyme-adapter-react-16";

Enzyme.configure({ adapter: new Adapter() });

const homeInfoData = {
  name: "Kingston City",
  abbr: "KC",
  color: "#d6211c",
  borderColor: "#de4d49"
};

const awayInfoData = {
  name: "Green Gully",
  abbr: "GG",
  color: "#eee",
  borderColor: "#ccc"
};

const scoreData = { home: 4, away: 0 };

const statData = { home: "Red Cards 0" };

const animationMain = [{ animation: "main", delay: 0 }];

const animationStat = [
  { animation: "main", delay: 0 },
  { animation: "teamStat", delay: 0.6 }
];

const setupEmpty = () => {
  const props = {
    home: homeInfoData,
    away: awayInfoData,
    score: scoreData,
    stat: statData,
    animation: []
  };
  const sWrapper = shallow(<ScoreBoard {...props} />);
  const mWrapper = mount(<ScoreBoard {...props} />);
  return {
    props,
    sWrapper,
    mWrapper
  };
};

const setupMain = () => {
  const props = {
    home: homeInfoData,
    away: awayInfoData,
    score: scoreData,
    stat: statData,
    animation: animationMain
  };
  const sWrapper = shallow(<ScoreBoard {...props} />);
  const mWrapper = mount(<ScoreBoard {...props} />);
  return {
    props,
    sWrapper,
    mWrapper
  };
};

const setupStat = () => {
  const props = {
    home: homeInfoData,
    away: awayInfoData,
    score: scoreData,
    stat: statData,
    animation: animationStat
  };
  const sWrapper = shallow(<ScoreBoard {...props} />);
  const mWrapper = mount(<ScoreBoard {...props} />);
  return {
    props,
    sWrapper,
    mWrapper
  };
};

describe("Main Sector Props", () => {
  const { sWrapper, mWrapper, props } = setupMain();

  it("get child component MainSection's length", () => {
    expect(sWrapper.find(MainSection).length).toBe(1);
    expect(mWrapper.find(MainSection).length).toBe(1);
  });

  test("get board-grid class", () => {
    expect(sWrapper.find(".board-grid").exists());
    expect(mWrapper.find(".board-grid").exists());
  });

  // Snapshot
  it("Main Section Snapshot", () => {
    expect(toJson(sWrapper)).toMatchSnapshot();
  });
});

describe("Stat Sector Props", () => {
  const { sWrapper, mWrapper, props } = setupStat();

  it("get child component MainSection's length", () => {
    expect(sWrapper.find(TeamStat).length).toBe(1);
    expect(mWrapper.find(TeamStat).length).toBe(1);
  });

  test("get board-grid class", () => {
    expect(sWrapper.find(".board-grid").exists());
    expect(mWrapper.find(".board-grid").exists());
  });

  // Snapshot
  it("Stat Section Snapshot", () => {
    expect(toJson(sWrapper)).toMatchSnapshot();
  });
});

describe("Empty Props", () => {
  const { sWrapper, mWrapper, props } = setupEmpty();

  it("get child component MainSection's length", () => {
    expect(sWrapper.find(TeamStat).length).toBe(1);
    expect(mWrapper.find(TeamStat).length).toBe(1);
  });

  test("get board-grid class", () => {
    expect(sWrapper.find(".board-grid").exists());
    expect(mWrapper.find(".board-grid").exists());
  });

  // Snapshot
  it("Empty Section Snapshot", () => {
    expect(toJson(sWrapper)).toMatchSnapshot();
  });
});
