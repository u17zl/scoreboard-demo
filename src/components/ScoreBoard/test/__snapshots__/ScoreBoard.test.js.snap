// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Empty Props Empty Section Snapshot 1`] = `
<Fragment>
  <div
    className="board-grid"
  >
    <MainSection
      away={
        Object {
          "abbr": "GG",
          "borderColor": "#ccc",
          "color": "#eee",
          "name": "Green Gully",
        }
      }
      delay={0}
      home={
        Object {
          "abbr": "KC",
          "borderColor": "#de4d49",
          "color": "#d6211c",
          "name": "Kingston City",
        }
      }
      score={
        Object {
          "away": 0,
          "home": 4,
        }
      }
      show={false}
      statShow={false}
    />
    <TeamStat
      away={
        Object {
          "abbr": "GG",
          "borderColor": "#ccc",
          "color": "#eee",
          "name": "Green Gully",
        }
      }
      delay={0}
      home={
        Object {
          "abbr": "KC",
          "borderColor": "#de4d49",
          "color": "#d6211c",
          "name": "Kingston City",
        }
      }
      isBoth={false}
      show={false}
      stat={
        Object {
          "home": "Red Cards 0",
        }
      }
    />
  </div>
</Fragment>
`;

exports[`Main Sector Props Main Section Snapshot 1`] = `
<Fragment>
  <div
    className="board-grid"
  >
    <MainSection
      away={
        Object {
          "abbr": "GG",
          "borderColor": "#ccc",
          "color": "#eee",
          "name": "Green Gully",
        }
      }
      delay={0}
      home={
        Object {
          "abbr": "KC",
          "borderColor": "#de4d49",
          "color": "#d6211c",
          "name": "Kingston City",
        }
      }
      score={
        Object {
          "away": 0,
          "home": 4,
        }
      }
      show={false}
      statShow={false}
    />
    <TeamStat
      away={
        Object {
          "abbr": "GG",
          "borderColor": "#ccc",
          "color": "#eee",
          "name": "Green Gully",
        }
      }
      delay={0}
      home={
        Object {
          "abbr": "KC",
          "borderColor": "#de4d49",
          "color": "#d6211c",
          "name": "Kingston City",
        }
      }
      isBoth={false}
      show={false}
      stat={
        Object {
          "home": "Red Cards 0",
        }
      }
    />
  </div>
</Fragment>
`;

exports[`Stat Sector Props Stat Section Snapshot 1`] = `
<Fragment>
  <div
    className="board-grid"
  >
    <MainSection
      away={
        Object {
          "abbr": "GG",
          "borderColor": "#ccc",
          "color": "#eee",
          "name": "Green Gully",
        }
      }
      delay={0}
      home={
        Object {
          "abbr": "KC",
          "borderColor": "#de4d49",
          "color": "#d6211c",
          "name": "Kingston City",
        }
      }
      score={
        Object {
          "away": 0,
          "home": 4,
        }
      }
      show={false}
      statShow={false}
    />
    <TeamStat
      away={
        Object {
          "abbr": "GG",
          "borderColor": "#ccc",
          "color": "#eee",
          "name": "Green Gully",
        }
      }
      delay={0}
      home={
        Object {
          "abbr": "KC",
          "borderColor": "#de4d49",
          "color": "#d6211c",
          "name": "Kingston City",
        }
      }
      isBoth={false}
      show={false}
      stat={
        Object {
          "home": "Red Cards 0",
        }
      }
    />
  </div>
</Fragment>
`;
